DEFRA/waste-permits
# Feature/we 587 front end validation

The purpose of this PR is to implement a working example of front end validation in our application.

https://eaflood.atlassian.net/browse/WE-587

Joi is our chosen validation framework: https://github.com/hapijs/joi

Validation has been added to our example 'Site' page. To make the validation more realistic I added a new 'Another Name' field to the page. This allows us to demonstrate having more than one error on the page.

A new Handlebars helper (getFormGroupClass.js) was added that allows a class in the HTML to be conditionally set based on whether or not the corresponding field contains a validation error.

This PR also includes the following improvements:

- Page heading and title refactoring
- Refactoring of our routes into the Constants file

To test:

I added the following validation rules to the site name and other name fields:

- Mandatory
- Must be less than 170 characters
- Cannot contain any of the following characters:  ^ | _ ~ ¬ `

So if you try to submit the Site page with fields containing any of these issues you should see the coresponding validation error messages appearing on the Site page.
